graphql_compiler/macros/macro_edge/directives.py

```diff
@@ -36,9 +41,26 @@
     ],
 )
 
-
 MACRO_EDGE_DIRECTIVES = (
     MacroEdgeDirective,
     MacroEdgeDefinitionDirective,
     MacroEdgeTargetDirective,
 )
+
+MACRO_EDGE_ALLOWED_SUBDIRECTIVES = frozenset({
+    FilterDirective.name,
+    TagDirective.name,
+    OptionalDirective.name,
+    RecurseDirective.name,
+    MacroEdgeDefinitionDirective.name,
+})
+
+MACRO_EDGE_DISALLOWED_SUBDIRECTIVES = frozenset({
+    OutputDirective.name,
+    OutputSourceDirective.name,
+})
+
+MACRO_EDGE_DEFINITION_DIRECTIVES = frozenset({
+    MacroEdgeDefinitionDirective,
+    MacroEdgeTargetDirective
+})
```

Comment:
It would be nice to document in one line what these mean. In what context are they allowed/disallowed? Someone might read this as "directives you are allowed to use on usage of macro edges in queries". See `_check_that_expansion_directives_are_supported` in `macro_expansion.py`.